src/pwm.rs

```diff
@@ -31,16 +31,104 @@
     let (c0, c1, c2, c3) = device.TIM2.pwm(
         pins,
         &mut afio.mapr,
-        100.hz()
+        100.hz(),
         clocks,
-        &mut rcc.apb
+        &mut rcc.apb1
     );
 
     // Set the duty cycle of channel 0 to 50%
     c0.set_duty(c0.get_max_duty() / 2);
     // PWM outputs are disabled by default
     c0.enable()
   ```
+
+  ## Usage for custom channel combinations
+
+  Note that crate itself defines only basic channel combinations for default AFIO remappings,
+  where all the channels are enabled. Meanwhile it is possible to configure PWM for any custom
+  selection of channels. The `Pins` trait shows the mapping between timers, output pins and
+  channels. So this trait needs to be implemented for the custom combination of channels and
+  AFIO remappings. However minor additional efforts are needed since it is not possible to
+  implement a foreign trait for a foreign type. The trick is to use the newtype pattern.
+
+  The first example selects PB5 channel for TIM3 PWM output:
+
+  ```
+  struct MyChannels(PB5<Alternate<PushPull>>);
+
+  impl Pins<TIM3>  for MyChannels {
+    const REMAP: u8 = 0b10; // use TIM3 AFIO remapping for PB4, PB5, PB0, PB1 pins
+    const C1: bool = false;
+    const C2: bool = true;  // use channel C2
+    const C3: bool = false;
+    const C4: bool = false;
+    type Channels = Pwm<TIM3, C2>;
+  }
+  ```
+
+  The second example selects PC8 and PC9 channels for TIM3 PWM output:
+
+  ```
+  struct MyChannels(PB5<Alternate<PushPull>>);
+
+  impl Pins<TIM3>  for MyChannels {
+    const REMAP: u8 = 0b11; // use TIM3 AFIO remapping for PC6, PC7, PC8, PC9 pins
+    const C1: bool = false;
+    const C2: bool = false;
+    const C3: bool = true;  // use channel C3
+    const C4: bool = true;  // use channel C4
+    type Channels = (Pwm<TIM3, C3>, Pwm<TIM3, C4>);
+  }
+  ```
+
+  REMAP value and channel pins should specified according to the stm32f1xx specification,
+  namely according to the chapter about AFIO and timer alternate function remappings.
```

Comment:
It would be nice to have a chapter/section number here

Reply:
Done: specified AFIO/TIM section from TRM RM0008 Rev 20.